viewing_party/party.py

```diff
@@ -1,23 +1,125 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    movies_info = {}
+    movies_info['title'] = title
+    movies_info['genre'] = genre
+    movies_info['rating'] = rating
+    if not title or not genre or not rating:
+        return None
+    return movies_info
+
+def add_to_watched(user_data, movie):
+    user_data['watched'].append(movie)
+    return user_data
+
+
+def add_to_watchlist(user_data, movie):
+    user_data['watchlist'].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data['watchlist']:
+        if movie['title'] == title:   
+            user_data['watchlist'].remove(movie)
+            user_data['watched'].append(movie)
+            # return user_data

```

Comment:
It would actually make sense to uncomment this! 😄 

[warning] Here we are iterating over the user's watchlist, and also modifying it (`remove`). We should avoid modifying a list while iterating over it to avoid problems with the iteration. _If_ we stop iterating as soon as we make a change, it would still be safe (though anyone else reading the code would need to convince themselves of this). However, depending on our data restrictions, this code could end up modifying the list multiple times if there is more than one movie with the same title.

To ensure that only one change is made, we could break out as soon as we find the movie and move it, either with a `break` statement or by doing `return user_data` like you have here commented out. Again, this is safe from an iteration perspective, but may require other team members to spend time convincing themselves of this. We might instead decide to think of this as 2 steps. 1. Find the movie to move by iterating through the list, and 2. actually move the move (outside the list). This might be more clear that there are no negative modifying while iterating shenanigans!